doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -195,19 +285,548 @@ Mail chapel-developers@ for review and merge
+
+If your work will require committing any third-party code that you are not
+developing yourself (or code that you've developed as a standalone package),
+alert the chapel-developers_ mailing list of this as, presently, such code
+packages must be approved by Cray leadership before being committed.
+
+.. _Testing guidance:
+
+Testing your patch
+++++++++++++++++++
+* Changes to the Chapel implementation should not cause regressions. Developers
+  are responsible for doing a degree of testing that's appropriate for their
+  change (described in the following bullets) and then can rely on nightly
+  regression testing to worry about the full cross-product of configurations.
+
+  * At a minimum, patches should pass correctness testing for the full test/
+    directory hierarchy for:
+
+    * ``CHPL_*_PLATFORM=linux64``
+    * ``CHPL_*_COMPILER=gnu``
+    * ``CHPL_COMM=none``
+    * ``CHPL_TASKS=<default>``
+
+  * Most developers will start by focusing on a subdirectory of tests that
+    exercise the features they changed, or test/release/ as a suite of tests
+    that exercises a rich and important slice of the language.
+
+  * Changes that are likely to affect multi-locale executions should also be
+    tested against tests that exercise multi-locale capabilities with
+    ``CHPL_COMM=gasnet``.  A common subset is: ``test/release/``,
+    ``test/multilocale/``, and ``test/distributions/``.
+
+  * Changes that are likely to cause portability issues should be tested against
+    different platforms and compilers to avoid fallout in the nightly testing to
+    the extent possible.
+
+* Note that the quickest way to do testing is to use the parallel testing system
+  across a large number of workstations.  If you have limited testing resources
+  available to you, you can request that a member of the core Chapel team help.
+
+.. _Test System: https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/bestPractices/TestSystem.rst
+
+.. _WIP PR:
+
+Work-in-progress pull requests (WIP PRs)
+++++++++++++++++++++++++++++++++++++++++
+
+This is a special kind of pull request that is not yet intended to be merged.
+Such PRs are created to take advantage of what the Github PR interface provides,
+such as public comment history and quick links between the WIP PR and other
+related issues and pull requests.  They allow the developer to get early
+feedback on a change.
+
+The status of the WIP PR should be clearly stated, including what steps need to
+be taken before the PR is ready for final review.
+
+It is generally advisable to "close" such PRs until they are ready for
+final review and testing, as their development can span a large period of time
+and would thus add clutter to the list of open PRs.
+
+It is perfectly acceptable to abandon such PRs (especially in favor of a cleaned
+up version of the code) when the git history becomes too large, so long as a
+link to the original PR is provided when the change is eventually merged.
```

Comment:
Looks good, thanks. I might say something about how linking to the original PR is good, to preserve the discussion. I'm not sure it's a "rule" at the same level of having signed a contributor agreement...